src/helm/benchmark/slurm_runner.py

```diff
@@ -0,0 +1,220 @@
+    cancel_slurm_job,
+    get_slurm_job_state,
+    SlurmJobState,
+    TERMINAL_SLURM_JOB_STATES,
+    FAILURE_SLURM_JOB_STATES,
+)
+from helm.common.general import ensure_directory_exists, asdict_without_nones
+from helm.common.hierarchical_logger import hlog
+
+
+@dataclass
+class SlurmRunnerSpec:
+    """Arguments to instantiate a SlurmRunner."""
+
+    execution_spec: ExecutionSpec
+    output_path: str
+    suite: str
+    skip_instances: bool
+    cache_instances: bool
+    cache_instances_only: bool
+    skip_completed_runs: bool
+    exit_on_error: bool
+
+    def to_kwargs(self):
+        return {field.name: getattr(self, field.name) for field in dataclasses.fields(self)}
+
+
+class SlurmRunner(Runner):
+    """Runner that runs the entire benchmark on Slurm."""
+
+    def __init__(
+        self,
+        execution_spec: ExecutionSpec,
+        output_path: str,
+        suite: str,
+        skip_instances: bool,
+        cache_instances: bool,
+        cache_instances_only: bool,
+        skip_completed_runs: bool,
+        exit_on_error: bool,
+    ):
+        self.slurm_runner_spec = SlurmRunnerSpec(
+            execution_spec=execution_spec,
+            output_path=output_path,
+            suite=suite,
+            skip_instances=skip_instances,
+            cache_instances=cache_instances,
+            cache_instances_only=cache_instances_only,
+            skip_completed_runs=skip_completed_runs,
+            exit_on_error=exit_on_error,
+        )
+        # Extra validation: Check that SlurmRunnerSpec can be used to initialize Runner.
+        super().__init__(**self.slurm_runner_spec.to_kwargs())
+        self.slurm_base_dir = os.path.join("slurm", datetime.now().isoformat(timespec="seconds"))
+
+    def run_all(self, run_specs: List[RunSpec]):
+        """Run the entire benchmark on Slurm, where each RunSpec is run in its own Slurm job.
+
+        This functions as a "manager" job that submits a Slurm worker job via sbatch for each RunSpec,
+        and then monitors all the Slurm worker."""
+        # This directory will be used for coordinating with workers
+        ensure_directory_exists(self.slurm_base_dir)
+
+        # Write the SlurmRunnerSpec to a file
+        slurm_runner_spec_json = json.dumps(asdict_without_nones(self.slurm_runner_spec), indent=2)
+        slurm_runner_spec_path = os.path.join(self.slurm_base_dir, "slurm_runner_spec.json")
+        hlog(f"Writing SlurmRunnerSpec to {slurm_runner_spec_path}")
+        with open(slurm_runner_spec_path, "w") as f:
+            f.write(slurm_runner_spec_json)
+
+        # Set up directories for RunSpecs and logs
+        run_specs_dir = os.path.join(self.slurm_base_dir, "run_specs")
+        ensure_directory_exists(run_specs_dir)
+        logs_dir = os.path.join(self.slurm_base_dir, "logs")
+        ensure_directory_exists(logs_dir)
+        run_name_to_slurm_job_id: Dict[str, int] = {}
+        run_name_to_slurm_job_state: Dict[str, str] = {}
```

Comment:
Could be nicer to make a dataclass that encompasses job id and job state and have one dictionary?